[DCOS-38141] Add SDK dashboard for scaletesting

elezar opened the pull request:
This PR adds a script to generate a Grafana dashboard for SDK-based services and adds an example dashboard.

Note that the dashboard "automatically" detects if SDK services are running and allows for the framework to be selected.

Note that `ROOT`/`MOUNT` volumes are not currently supported by this dashboard.

![screen shot 2018-06-13 at 17 03 01](https://user-images.githubusercontent.com/931393/41366370-280cb3c8-6f3c-11e8-96b0-b7dd80031e32.png)

porridge commented on scale-tests/sdk.dashboard.py:
```diff
@@ -0,0 +1,266 @@
+    return base
+
+
+def metric(name: str, selection: str) -> str:
```
Consider making selection a dict for a more pythonic invocation syntax.

porridge commented on scale-tests/sdk.dashboard.py:
```diff
@@ -0,0 +1,266 @@
+installed on a Cluster.
+"""
+
+from grafanalib.core import *
```
Please enumerate the pieces you need. `*` is considered bad style because it's a maintenance headache.

porridge commented on scale-tests/sdk.dashboard.py:
```diff
@@ -0,0 +1,266 @@
+    )
+
+
+# Generate the Grafana dashboard.
```
This comment is not very useful. It's clear from the subsequent line that you're creating a dashboard...

elezar commented on scale-tests/sdk.dashboard.py:
````diff
@@ -0,0 +1,266 @@
+```bash
+pip install grafanalib
+
+generate-dashboard -o sdk.json sdk.dashboard.py
````
From: https://github.com/weaveworks/grafanalib#generating-dashboards
> If you save the above as frontend.dashboard.py (the suffix must be .dashboard.py), you can then generate the JSON dashboard with:

porridge commented:
Ah, that's interesting. Mind linking to that page there?

elezar commented:
Thanks @porridge and @samvantran for the review. I have addressed the issues you raise.

@samvantran with regards to:
> I ran generate-dashboard -o sdk.json sdk.dashboard.py and uploaded the generated json to Grafana and this is what I saw:

You need to select a `framework` from the drop-down list at the top of the page. It could be that the metrics are not present to populate the list, but this has worked every time that I have tested it.
